lib/key-agreement-key.js

```diff
@@ -83,13 +72,9 @@ exports.deriveSecret = async ({key, operation}) => {
       `The given public key type "${publicKey.type}" does not match the ` +
       `key agreement key's ${key.type}.`);
   }
-  const privateKey = base58.decode(key.privateKeyBase58);
-  const remotePublicKey = base58.decode(publicKey.publicKeyBase58);
-  const secret = await _deriveSecret({privateKey, remotePublicKey});
-  return {secret: base64url.encode(secret)};
-};
 
-async function _deriveSecret({privateKey, remotePublicKey}) {
-  // `scalarMult` takes secret key as param 1, public key as param 2
-  return nacl.scalarMult(privateKey, remotePublicKey);
-}
+  const keyPair = await X25519KeyAgreementKey2020.from(key);
+  const secret = await keyPair.deriveSecret({publicKey});
+
+  return {secret};
```

Comment:
The `secret` used to be base64url encoded before being returned. When I looked at the source for this new method, it does not do that (and it shouldn't), but I did notice that the docs are wrong:

https://github.com/digitalbazaar/x25519-key-agreement-key-2020/blob/main/lib/X25519KeyAgreementKey2020.js#L293

Since it says it returns a string when it should be a Uint8Array. Regardless, The KMS module API previously required a base64url value to be returned -- has that changed? If it hasn't and this is passing tests, please add a test so that it fails and then fix it so it does the encoding.
```suggestion
  const secret = await keyPair.deriveSecret({publicKey});

  return {secret};
```

Reply:
@dlongley ah, dammit, good catch. I was going by that jsdoc on line 293, yeah. 
See, now if we were using _TypeScript_, this wouldn't've happened :)

Reply:
> See, now if we were using TypeScript, this wouldn't've happened :)

I'm not even sure if that's true ... but if so, congrats, out of thousands of potential bugs, TypeScript may have saved us here! And the only price to pay to catch it would be the overhead of having all devs use it all the time, everywhere. :)

Reply:
Addressed, added failing test & fixed it.

Reply:
Thanks!